tests/image/image_data_generator_test.py

```diff
@@ -442,5 +442,34 @@ def test_random_transforms():
     assert transform_dict['brightness'] is None
 
 
+def test_fit_rescale(all_test_images):
+    rescale = 1. / 255
+
+    for test_images in all_test_images:
+        img_list = []
+        for im in test_images:
+            img_list.append(utils.img_to_array(im)[None, ...])
+        images = np.vstack(img_list)
+
+        # featurewise_center and featurewise_std_normalization test
+        generator = image_data_generator.ImageDataGenerator(
+            rescale=rescale,
+            featurewise_center=True,
+            featurewise_std_normalization=True)
+        generator.fit(images)
+        batch = generator.flow(images, batch_size=8).next()
+        assert abs(np.mean(batch)) < 1e-3
+        assert abs(1 - np.std(batch)) < 1e-3
+
+        # zca_whitening test
+        generator = image_data_generator.ImageDataGenerator(
+            rescale=rescale,
+            featurewise_center=True,
+            zca_whitening=True)
+        generator.fit(images)
+        batch = generator.flow(images, batch_size=8).next()
+        assert np.max(np.abs(batch)) <= 1
```

Comment:
please check thee mean here as well.

Reply:
I digged deeper into the ZCA whitening topic and proposed a better test condition in the latest commit. Basically, after applying whitening, we obtain a new matrix **_Y_** (**_Y_** = **_WX_**, where **_W_** is a decorrelating matrix) where **_YY_**<sup>T</sup> is a diagonal matrix such that **_YY_**<sup>T</sup> = _(n-1)_**_I_** (_n_ is a number of data samples).

(More about this can be found in [Learning Multiple Layers of Features from Tiny Images](http://www.cs.toronto.edu/~kriz/learning-features-2009-TR.pdf) in Appendix A The ZCA whitening transformation)